static-www/scripts/generate_www.pl

```diff
@@ -310,11 +311,17 @@ sub fetch_chart_data {
 }
 
 sub copy_src {
-    say "copy_src() is running: cp $src_in/* $src_out/";
-    # huh. Couldn't get File::Copy working quickly...
-    # copy("static-www/src/*", $out_root) or die "Can't copy $_: $!" for <*.txt>;
-    # So I guess we'll make system() calls, which is Bad(tm)?
-    system("mkdir $src_out") unless (-d "$src_out");
-    system("cp $src_in/* $src_out/");  # or die "$! $src_in/* -> $src_out/";
-                                        # ^ die works, but throws an error at the end
+    say "copy_src() is running: copying $src_in/* to $src_out/";
+    make_path($src_out) unless -d $src_out;
+    opendir(my $dh, $src_in) or die "Cannot open $src_in: $!";
+    while (my $file = readdir($dh)) {
+        next if $file =~ /^\./; # Skip . and ..
+        my $src_file = "$src_in/$file";
+        my $dst_file = "$src_out/$file";
+        if (-f $src_file) {
+            copy($src_file, $dst_file) or die "Cannot copy $src_file to $dst_file: $!";
+            say "Copied $src_file to $dst_file";
+        }
+    }
+    closedir($dh);
```

Comment:
Ha! Congrats! I coded Perl for 20 years, so I tried to do this "the Right Way(tm)" first, but got frustrated and punted. :)

Reply:
😄